includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1951,7 +1961,9 @@ private function check_attr_spec_rule_allowed_protocol( DOMElement $node, $attr_
 					if ( $node->hasAttribute( $alternative_name ) ) {
 						foreach ( $this->extract_attribute_urls( $node->getAttributeNode( $alternative_name ), $attr_name ) as $url ) {
 							$url_scheme = $this->parse_protocol( $this->normalize_url_from_attribute_value( $url ) );
-							if ( isset( $url_scheme ) && ! in_array( strtolower( $url_scheme ), $attr_spec_rule[ AMP_Rule_Spec::VALUE_URL ][ AMP_Rule_Spec::ALLOWED_PROTOCOL ], true ) ) {
+							if ( $this->args['allow_localhost_http_protocol'] && self::LOCALHOST === wp_parse_url( $url, PHP_URL_HOST ) ) {
+								return AMP_Rule_Spec::PASS;
```

Comment:
Is there a missing test case here? Perhaps there is no case which causes this code to be executed.

Reply:
Yeah, missing for `alternative_names`. Added in https://github.com/ampproject/amp-wp/pull/7508/commits/c0539bea39c245c1a8420e671075965b5928721f